src/ECharts/Coords.purs

```diff
@@ -0,0 +1,106 @@
+module ECharts.Coords where
+
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Combinators
+
+import Data.Maybe
+import Data.StrMap (fromList)
+
+data XPos = XLeft
+          | XRight
+          | XCenter
+          | X Number
+
+
+data YPos = YTop
+          | YBottom
+          | YCenter
+          | Y Number
+
+
+instance xPosShow :: Show XPos where
+  show pos = case pos of
```

Comment:
Generally, `Show` instances are supposed to render the data as code, which can reconstruct the data values. This can be useful during debugging. Since these instances don't render the values as code, it's best to turn them into ordinary functions (or, if the intent is rendering to JSON, make EncodeJson instances for them).